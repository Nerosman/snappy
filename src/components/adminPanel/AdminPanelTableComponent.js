import React from 'react';
import {Table} from "reactstrap"

export const AdminPanelTableComponent = ({orderList}) => {
    return orderList.length > 0 ? <Table responsive>
        <thead>
        <tr>
            <th>#</th>
            <th>First name</th>
            <th>Second name</th>
            <th>Address</th>
            <th>City, State, Zip</th>
            <th>Email</th>
            <th>Phone</th>
            <th>Special notes</th>
        </tr>
        </thead>
        <tbody>
        {orderList.map((order, index) => {
            return (
                <tr key={index}>
                    <td>{index + 1}</td>
                    <td>{order.firstName}</td>
                    <td>{order.secondName}</td>
                    <td>{order.address1}</td>
                    <td>{order.address2}</td>
                    <td>{order.email}</td>
                    <td>{order.phone}</td>
                    <td>{order.specialNotes}</td>
                </tr>
            );
        })}
        </tbody>
    </Table> : <div>There is no data to display</div>
};