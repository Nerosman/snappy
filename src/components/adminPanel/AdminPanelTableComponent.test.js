import ReactDOM from "react-dom";
import React from "react";
import {AdminPanelTableComponent} from "./AdminPanelTableComponent";
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe("component <AdminPanelTableComponent />", ()=>{
    it('render div into AdminPanelTableComponent without table', () => {
        const div = document.createElement('div');
        ReactDOM.render(<AdminPanelTableComponent />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('snapshot adminPanel', () => {
        const wrapper = shallow(<AdminPanelTableComponent/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('render AdminPanelTableComponent with table', () => {
        const wrapper = shallow(<AdminPanelTableComponent orderList={[{firstName: "Bohdan", secondName: "Yuzyfyshyn", address1: "Kn. Olgy 12", address2: "Lviv", email: "nerosman2601@gmail.com", phone: "0323212323"}]}/>);
        expect(wrapper)
    });

});
