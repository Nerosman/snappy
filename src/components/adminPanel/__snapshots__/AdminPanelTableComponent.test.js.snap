// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <AdminPanelTableComponent /> snapshot adminPanel 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminPanelTableComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "There is no data to display",
    },
    "ref": null,
    "rendered": "There is no data to display",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "There is no data to display",
      },
      "ref": null,
      "rendered": "There is no data to display",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
